app/lib/dns-server.ts

```diff
@@ -0,0 +1,239 @@
+                },
+              ],
+            },
+          },
+        ],
+      },
+      HostedZoneId: hostedZoneId,
+    });
+    const response = await route53Client.send(command);
+
+    if (!response.ChangeInfo?.Id) {
+      throw new Error(`Could not create a record`);
+    } else {
+      return response.ChangeInfo?.Id;
+    }
+  } catch (error) {
+    throw new Error(
+      `Something went wrong while creating resource record: ${error}`
+    );
+  }
+};
+
+export const updateRecord = async (
+  type: RecordType,
+  name: string,
+  value: string,
+  hostedZoneId: string
+): Promise<string> => {
+  try {
+    const command = new ChangeResourceRecordSetsCommand({
+      ChangeBatch: {
+        Changes: [
+          {
+            Action: "UPSERT",
+            ResourceRecordSet: {
+              Name: name,
+              Type: type,
+              ResourceRecords: [
+                {
+                  Value: value,
+                },
+              ],
+            },
+          },
+        ],
+      },
+      HostedZoneId: hostedZoneId,
+    });
+    const response = await route53Client.send(command);
+
+    if (!response.ChangeInfo?.Id) {
+      throw new Error(`Could not update the record`);
+    } else {
+      return response.ChangeInfo?.Id;
+    }
+  } catch (error) {
+    throw new Error(
+      `Something went wrong while updating resource record: ${error}`
+    );
+  }
+};
+
+export const deleteRecord = async (
+  type: RecordType,
+  name: string,
+  value: string,
+  hostedZoneId: string
+): Promise<string> => {
+  try {
+    const command = new ChangeResourceRecordSetsCommand({
+      ChangeBatch: {
+        Changes: [
+          {
+            Action: "DELETE",
+            ResourceRecordSet: {
+              Name: name,
+              Type: type,
+              ResourceRecords: [
+                {
+                  Value: value,
+                },
+              ],
+            },
+          },
+        ],
+      },
+      HostedZoneId: hostedZoneId,
+    });
+
+    const response = await route53Client.send(command);
+
+    if (!response.ChangeInfo?.Id) {
+      throw new Error("Could not delete the record");
+    } else {
+      return response.ChangeInfo?.Id;
+    }
+  } catch (error) {
+    throw new Error(
+      `Something went wrong while deleting resource record: ${error}`
+    );
+  }
+};
+
+export const listRecords = async (
+  hostedZoneId: string
+): Promise<ResourceRecordSet[]> => {
+  try {
+    const command = new ListResourceRecordSetsCommand({
+      HostedZoneId: hostedZoneId,
+    });
+    const response = await route53Client.send(command);
+
+    if (!response.ResourceRecordSets) {
+      throw new Error("Could not get resource records");
+    } else {
+      return response.ResourceRecordSets;
+    }
+  } catch (error) {
+    throw new Error(
+      `Something went wrong while fetching resource record: ${error}`
+    );
+  }
+};
+
+export const getChangeStatus = async (requestId: string) => {
```

Comment:
`requestId` should be `changeId` I think.  Also, does Route53 require you to strip the leading `change/` from the id or will it work if you use exactly what it responds with on the `ChangeInfo.Id`?

Reply:
I don't think it works with `/change/` but I will confirm it. Thanks

Reply:
If that's the case, we should strip the string prefix like you do with the hosted zone id result.